orttraining/orttraining/python/orttraining_pybind_state.cc

```diff
@@ -840,24 +840,20 @@ void addObjectMethodsForTraining(py::module& m, ExecutionProviderRegistrationFn
           ORT_THROW_IF_ERROR(onnxruntime::training::api::SaveCheckpoint(trainable_tensor_protos,
                                                                         non_trainable_tensor_protos, checkpoint_path));
         });
-  m.def("load_checkpoint",
-        [](const std::string& checkpoint_path) {
+  m.def("load_checkpoint_to_model",
+        [](const std::string& checkpoint_path, py::bytes& serialized_model) {
           std::vector<TensorProto> tensor_protos;
-          ORT_THROW_IF_ERROR(onnxruntime::training::api::LoadCheckpoint(checkpoint_path, tensor_protos));
-          std::vector<py::bytes> tensor_protos_pybytes(tensor_protos.size());
-
-          const auto parse_tensor_proto_to_pybytes =
-              [](std::vector<py::bytes>& tensor_protos_pybytes, const std::vector<TensorProto>& tensor_protos) {
-                for (size_t i = 0; i < tensor_protos.size(); ++i) {
-                  std::string tensor_proto_str;
-                  tensor_protos[i].SerializeToString(&tensor_proto_str);
-                  tensor_protos_pybytes[i] = tensor_proto_str;
-                }
-              };
+          ONNX_NAMESPACE::ModelProto model_proto;
+
+          std::istringstream buffer(serialized_model);
+          Status status = Model::Load(buffer, &model_proto);
+          status = onnxruntime::training::api::LoadCheckpointToModel(checkpoint_path, model_proto);
```

Comment:
you are overwriting the status without checking whether Model load was successful

Reply:
resolved by using ORT_THROW_IF_ERROR on both Load model and LoadCheckpointToModel.